odoo/tools/xml_utils.py

```diff
@@ -263,15 +263,25 @@ def validate_xml_from_attachment(env, xml_content, xsd_name, reload_files_functi
     """Try and validate the XML content with an XSD attachment.
     If the XSD attachment cannot be found in database, (re)load it.
 
+    A skip_xsd key can be provided in the context in order to skip the XSD validation.
+    This should be used during tests to avoid loading XSD files (and making http requests every time).
+
     :param odoo.api.Environment env: environment of calling module
     :param xml_content: the XML content to validate
     :param xsd_name: the XSD file name in database
+    :param reload_files_function: function that will be called to try and (re)load XSD files
     :return: the result of the function :func:`odoo.tools.xml_utils._check_with_xsd`
     """
+    if env.context.get('skip_xsd', False):
+        return
```

Comment:
That makes no sense. If the caller wants the check to be conditional they can just call it conditionally.

Reply:
It's not always that easy. Most of the time, XSD validation is called during some sort of `get_xml` method after the XML generation, or even before posting an invoice. This means we have no real control when calling those methods during tests, and it's easier to call them with `with_context` rather than with a condition.
[Example 1](https://github.com/odoo-dev/enterprise/blob/master-xsd-validation-anpp/l10n_cl_edi/tests/test_l10n_cl_account_move_reversal.py#L34)
[Example 2](https://github.com/odoo-dev/enterprise/blob/master-xsd-validation-anpp/l10n_no_saft/tests/test_no_saft_report.py#L84)